go/store/datas/pull.go

```diff
@@ -102,56 +115,108 @@ func clone(ctx context.Context, srcTS, sinkTS nbs.TableFileStore, eventCh chan<-
 		return err
 	}
 
+	desiredFiles, fileIDToTF := mapTableFiles(tblFiles)
+
 	if eventCh != nil {
 		eventCh <- TableFileEvent{Listed, tblFiles}
 	}
 
-	// should parallelize at some point
-	for _, tblFile := range tblFiles {
-		if eventCh != nil {
-			eventCh <- TableFileEvent{DownloadStart, []nbs.TableFile{tblFile}}
-		}
+	i := 0
+	op := func() error {
+		var err error
+		for i < len(desiredFiles) {
+			fileID := desiredFiles[i]
+			tblFile, ok := fileIDToTF[fileID]
 
-		err := func() (err error) {
-			var rd io.ReadCloser
-			rd, err = tblFile.Open()
+			if !ok {
```

Comment:
If the set of files in fileIDToTF can't change from reassignment below, then I don't understand the point of this check or the conjoin comment. It seems like this is unreachable.

Reply:
fileIDToTF can change.  For instance if there was a conjoin that ran between the first call to srcTS.Sources and the second then the fileIDToTF map would have different fileIDs.